sqlmesh/core/model/definition.py

```diff
@@ -1784,6 +1785,84 @@ class AuditResult(PydanticModel):
     blocking: bool = True
 
 
+def create_model_blueprints(
+    gateway: t.Optional[str | exp.Expression],
+    blueprints: t.Any,
+    get_variables: t.Callable[[t.Optional[str]], t.Dict[str, str]],
+    loader: t.Callable[..., Model],
+    **loader_kwargs: t.Any,
+) -> t.List[Model]:
+    path = loader_kwargs.get("path")
+    dialect = loader_kwargs.get("dialect")
+
+    if not blueprints:
+        blueprints = [None]
+    elif isinstance(blueprints, exp.Paren):
+        blueprints = [blueprints.unnest()]
+    elif isinstance(blueprints, (exp.Tuple, exp.Array)):
+        blueprints = blueprints.expressions
+    elif not isinstance(blueprints, list):
+        # TODO: put docs link here
+        raise ConfigError(
+            f"Invalid 'blueprints' property '{blueprints}' at '{path}', please refer to <link>."
+        )
+
+    model_blueprints: t.List[Model] = []
+    for blueprint in blueprints:
+        if not blueprint:
+            variables = {}
+        elif isinstance(blueprint, exp.Paren):
+            blueprint = blueprint.unnest()
+            variables = {blueprint.left.name: blueprint.right.sql(dialect=dialect)}
+        elif isinstance(blueprint, (exp.Tuple, exp.Array)):
+            variables = {e.left.name: e.right.sql(dialect=dialect) for e in blueprint.expressions}
+        elif isinstance(blueprint, dict):
+            variables = blueprint
+        else:
+            # TODO: put docs link here
+            raise ConfigError(
+                f"Invalid blueprint value: {blueprint} at '{path}', please refer to <link>."
+            )
+
+        if isinstance(gateway, d.MacroVar):
+            gateway_name = variables.get(gateway.name)
+        elif isinstance(gateway, str) and gateway.startswith("@"):
+            gateway_name = variables.get(gateway[1:])
+        else:
+            gateway_name = None
```

Comment:
What if the gateway is a macro call and not a variable?

Reply:
Good question. I guess that would result in a failed lookup because of the `(` and `)` in the call text, granted though that this is somewhat implicit. Other than that, I think it should cover the use case we're interested in, i.e. rendering blueprint variables.

Reply:
Addressed, please see updated logic & test.

Reply:
looks good